app/plugin.go

```diff
@@ -460,6 +463,25 @@ func (a *App) DeletePublicKey(file string) *model.AppError {
 	return nil
 }
 
+// VerifyPlugin verifies plugin signature
+func (a *App) VerifyPlugin(plugin, signature io.Reader) *model.AppError {
+	publicKeys, appErr := a.GetPluginPublicKeys()
+	if appErr != nil {
+		return appErr
+	}
+	for _, pk := range publicKeys {
+		pkBytes, appErr := a.GetPublicKey(pk)
+		if appErr != nil {
+			mlog.Error("Unable to get public key for ", mlog.String("filename", pk))
+		}
```

Comment:
Should it return if it was not able to read the public key?

Reply:
Actually, I'm not sure about that. Might happen that we were not able to get one of the public keys, but still were able to get a public key corresponding to the signature and were able to verify the plugin. In this case we will install plugin and output an error that there is something wrong with one of the public keys.